cranelift/codegen/src/egraph.rs

```diff
@@ -197,17 +202,20 @@ impl<'a> FuncEGraph<'a> {
                 let node = if is_readonly_load {
                     self.stats.node_created += 1;
                     self.stats.node_pure += 1;
-                    Node::Pure { op, args, types }
+                    Node::Pure {
+                        op,
+                        args,
+                        ty,
+                        arity: results.len() as u16,
```

Comment:
```suggestion
                        arity: {
                            debug_assert!(results.len() <= (u16::MAX as usize));
                            results.len() as u16
                        },
```

Reply:
Fixed! Actually used `u16::try_from(...).expect(...)` here but result is the same. And, factored out above if/else chain to use below as well.